crates/agent-sql/src/directives/beta_onboard.rs

```diff
@@ -41,39 +41,86 @@ pub async fn tenant_exists(
     Ok(exists.is_some())
 }
 
-/*
-TODO(johnny): A fast-follow will be inserting `ops/` specifications for the user:
-insert into live_specs (catalog_name, last_build_id, last_pub_id, spec, spec_type) values
-    ('ops/' || $2 || 'logs',  '00:00:00:00:00:00:00:00', '00:00:00:00:00:00:00:00', $4, 'collection'),
-    ('ops/' || $2 || 'stats', '00:00:00:00:00:00:00:00', '00:00:00:00:00:00:00:00', $5, 'collection')
-*/
-pub async fn provision_user(
-    user_id: Uuid,
-    tenant: &str,
+// ProvisionedTenant is the shape of a provisioned tenant.
+#[derive(Debug)]
+pub struct ProvisionedTenant {
+    // Draft into which provisioned catalog specs should be placed.
+    // It will be queued for publication upon the commit of this transaction.
+    pub draft_id: Id,
+}
+
+pub async fn provision_tenant(
+    accounts_user_email: &str,
     detail: Option<String>,
+    tenant: &str,
+    tenant_user_id: Uuid,
     txn: &mut sqlx::Transaction<'_, sqlx::Postgres>,
-) -> sqlx::Result<()> {
+) -> sqlx::Result<ProvisionedTenant> {
     let prefix = format!("{tenant}/");
 
-    sqlx::query!(
-        r#"with s1 as (
+    let provisioned = sqlx::query_as!(
+        ProvisionedTenant,
+        r#"with
```

Comment:
Are all of these CTEs the same as/equivalent to using a transaction? My concern would be leaving things in a broken state if something fails here, and we don't automatically rollback

Reply:
We're already in a transaction within this context, so they could safely be separate statements.

The CTE's are for tighter narrative structure / less boilerplate. It's slightly more efficient to boot but that doesn't matter here.